packages/headless/src/features/folding/folding-slice.test.ts

```diff
@@ -1,163 +1,144 @@
+  const part: Hierarchy = {name: root.result.title};
+  if (root.children.length) {
+    part.children = root.children.map((child) => extractHierarchy(child));
+  }
+  return part;
+}
 
-    expect(results).toEqual([expectedHierarchy, expectedOtherResult]);
+describe('folding slice', () => {
+  it('should be disabled by default', () => {
+    expect(foldingReducer(undefined, {type: ''}).enabled).toBeFalsy();
   });
 
-  it('should correctly fold new hierarchical results', () => {
-    const {
-      animalsResult,
-      relevantResult,
-      ballPythonsResult,
-      geckosResult,
-      mammalsResult,
-    } = getEncyclopediaCollection();
-
-    const foldedOtherResult: FoldedResult = {
-      ...buildMockResultWithFolding(),
-      children: [],
-    };
-
-    const initialState: FoldingState = {
-      ...getFoldingInitialState(),
-      enabled: true,
-      fields: {collection: 'collection', parent: 'parent', child: 'id'},
-      collections: [foldedOtherResult],
-    };
-
-    const action = fetchMoreResults.fulfilled(
-      buildMockSearch({
-        response: buildMockSearchResponse({
-          results: [relevantResult],
-        }),
-      }),
-      '',
-      null as never
-    );
-
-    const expectedHierarchy = {
-      ...animalsResult,
+  describe('with folding enabled', () => {
+    const testThreadHierarchy: Hierarchy = {
+      name: 'some-question',
       children: [
         {
-          ...relevantResult,
+          name: 'first-answer',
           children: [
-            {...ballPythonsResult, children: []},
-            {...geckosResult, children: []},
+            {
+              name: 'comment-questionning-the-answer',
+              children: [
+                {name: 'comment-justifying-the-answer'},
+                {
+                  name: 'some-deleted-answer',
+                  children: [{name: 'comment-that-has-no-context-anymore'}],
+                },
+              ],
+            },
           ],
         },
-        {...mammalsResult, children: []},
+        {
+          name: 'better-response',
+        },
       ],
     };
 
-    const results = foldingReducer(initialState, action).collections;
-
-    expect(results).toEqual([foldedOtherResult, expectedHierarchy]);
+    let state: FoldingState;
+
+    function dispatch(action: AnyAction) {
+      state = foldingReducer(state, action);
+    }
+
+    function dispatchSearch(results: ResultWithFolding[]) {
+      dispatch(
+        executeSearch.fulfilled(
+          buildMockSearch({
+            response: buildMockSearchResponse({
+              results,
+            }),
+          }),
+          '',
+          null as never
+        )
+      );
+    }
+
+    beforeEach(() => {
+      state = {
+        enabled: true,
+        collections: {},
+        fields: {collection: 'collection', parent: 'parent', child: 'id'},
+        filterFieldRange: -1234,
+      };
+    });
+
+    it('should correctly fold a single collection from the root', () => {
```

Comment:
For now, this is the only test; more are going to be added.

At first I tried to fold from a result deeper in the hierarchy, but `emulateAPIFolding` excluded a sibling result (which seems to be how the API behaves as well) and I wanted to keep the test short for now.